core/src/main/java/hudson/model/Executor.java

```diff
@@ -594,6 +653,32 @@ protected Object call(Object o, Method m, Object[] args) throws Throwable {
     }
 
     /**
+     * Finds the executor running a given process.
```

Comment:
Well you can still have `Executor executable = Executor.of(executable); assert executable.getCurrentExecutable() == executable` give an Assertion error` as the executable can finish concurrent with returning from the method.